### PR TITLE
Use yarn workspace

### PR DESCRIPTION
**Pull request by huozhi**
- [x] group core libraries
- [x] remove example development 
- [x] update contribution docs
- [x] simplify publish script
- [x] remove yarn link / yalc related tricks

### CODE REVIEW

File: yalc.lock [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "version": "v1",
+  "packages": {
+    "swr": {
+      "signature": "984b0beb41ffe50f87a7a947b8de4e6a"
+    }
+  }
+}
```
Comment (shuding) [inline yalc.lock:8]: What is this file for?

File: test/use-swr-focus.test.tsx [hunk 2]
```diff
@@ -195,11 +195,7 @@ describe('useSWR - focus', () => {
     await focusWindow()
     // wait for throttle intervals
     await act(() => sleep(150))
-    // trigger revalidation
-    await focusWindow()
-    await waitForNextTick()
-    await act(() => sleep(100))
-    await screen.findByText('data: 5')
+    await screen.findByText('data: 4')
```
Comment (shuding) [inline test/use-swr-focus.test.tsx:198]: Why changing this test?

**Review by shuding — COMMENTED** [review]
Can we rename the “package” directory to “src”?

File: package.json [hunk 3]
```diff
@@ -97,11 +66,9 @@
...
-    "typescript": "4.4.3",
-    "yalc": "1.0.0-pre.53"
-  },
-  "peerDependencies": {
-    "react": "^16.11.0 || ^17.0.0 || ^18.0.0"
+    "typescript": "4.4.3"
   },
-  "dependencies": {}
+  "dependencies": {
+    "next": "12.0.7"
```
Comment (shuding) [inline package.json:72]: Should this be in `devDependencies`?

**Comment by huozhi** [comment]
> Can we rename the “package” directory to “src”?
@shuding Choose `package` because it contains src, package.json and dist files (`package/dist`) like the previous swr package. `src` might be a little bit inaccurate? wdyt

File: test/use-swr-focus.test.tsx (hunk 2 above)
Comment (huozhi) [reply test/use-swr-focus.test.tsx:198]: This flaky test failed almost every time..

**Comment by shuding** [comment]
I'm not sure if `package` is the best name here — if the goal is to make development easier with examples, I think we can organize the files in a leaner way.

**Pull request closed without merging**